python/lsst/cp/pipe/ptc/cpPtcAdjustGainRatios.py

```diff
@@ -244,125 +285,190 @@ def run(self, *, exposures, input_ptc):
-                    binned["value"]/norm,
-                    np.array([]),
-                    constrain_zero=False,
-                    fit_centroid=False,
-                    fit_gradient=False,
-                    fp_centroid_x=0.0,
-                    fp_centroid_y=0.0,
-                )
-                p0 = fitter.compute_p0()
-                pars = fitter.fit(p0)
-
-                binned["value"] /= fitter.compute_model(pars)
-
-            # Fit gain ratios.
-            fitter = FlatGainRatioFitter(
-                exposure.getBBox(),
-                self.config.chebyshev_gradient_order,
-                binned["xd"],
-                binned["yd"],
-                binned["amp_index"],
-                binned["value"],
-                np.arange(len(ptc.ampNames)),
-                fixed_amp_index,
+    return ref_amp_index
+
+
+def _compute_gain_ratios(
+    detector,
+    binned,
+    fixed_amp_index,
+    do_remove_radial_gradient=True,
+    radial_gradient_n_spline_nodes=20,
+    chebyshev_gradient_order=1,
+    max_fractional_gain_ratio=0.05,
+    nsig_clip=5.0,
+    log=None,
+):
+    """Compute the gain ratios from a given non-gain-corrected exposure.
+
+    Parameters
+    ----------
+    detector : `lsst.afw.cameraGeom.Detector`
+        Detector object.
+    binned : `astropy.table.Table`
+        Table of binned values. Will contain ``xd``, ``yd`` (detector
+        bin positions); ``value`` (binned value); ``amp_index``
+        (index of the amplifiers).
+    fixed_amp_index : `int`
+        Use this amp index as the fixed point (gain ratio == 1.0).
+    do_remove_radial_gradient : `bool`, optional
+        Remove radial gradient before fitting amp gain ratios?
+    radial_gradient_n_spline_nodes : `int`, optional
+        Number of radial spline nodes for radial gradient.
+    chebyshev_gradient_order : `int`, optional
+        Order of chebyshev x/y polynomials to remove additional gradients.
+    max_fractional_gain_ratio : `int`, optional
+        Maximum fractional gain ratio to consider per-exposure. Any
+        amps with larger offset will be excluded from the gradient fit
+        and will have no corrections computed.
+    nsig_clip : `float`, optional
+        Number of sigma on distribution of gain ratios to clip when fitting
+        out Chebyshev gradient.
+    log : `logging.logger`, optional
+        Log object.
+
+    Returns
+    -------
+    amp_gain_ratios : `np.ndarray`
+        Amp gain ratios, relative to fixed amp.
+    """
+    log = log if log else logging.getLogger(__name__)
+
+    n_amp = len(detector)
+    amp_names = [amp.getName() for amp in detector]
+
+    # We iterate up to 8x to look for any bad amps (that have a gain offset
+    # fit greater than max_fractional_gain_ratio) and reject them
+    # from the fits.
+    bad_amps_converged = False
+    n_iter = 0
+    value_uncorrected = binned["value"].copy()
+    while not bad_amps_converged and n_iter < n_amp // 2:
```

Comment:
I'm not sure this handles a non-converging fit where `n_iter > ~8`.

Reply:
In that case it just stops.  But I don't think that I've ever seen that.  Also, this is not new code; it's just been factored out into its own function.